src/main/java/seedu/address/model/session/Session.java

```diff
@@ -50,12 +50,19 @@ public Fee getFee() {
      * @param otherSession the other session that is compared to.
      */
     public boolean isOverlapping(Session otherSession) {
-        SessionDate otherSessionDate = otherSession.getSessionDate();
+        SessionDate otherSessionStartDate = otherSession.getSessionDate();
+        SessionDate otherSessionEndDate = otherSessionStartDate.getEndSessionDate(otherSession.duration);
         SessionDate sessionStartDate = sessionDate;
         SessionDate sessionEndDate = sessionStartDate.getEndSessionDate(duration);
-        return (otherSessionDate.getDateTime().isEqual(sessionStartDate.getDateTime())
-                || otherSessionDate.getDateTime().isAfter(sessionStartDate.getDateTime()))
-                && otherSessionDate.getDateTime().isBefore(sessionEndDate.getDateTime());
+        boolean otherSessionOverlapsAfterSessionStarts = (
+                otherSessionStartDate.getDateTime().isEqual(sessionStartDate.getDateTime())
```

Comment:
not too sure if this condition is needed since this is already checked in `hasSession()` method.
Same for the other boolean condition below.

Reply:
Are you referring to line 58? I am aware of that and have thought about it and decided to add it in as well since it wouldn't make sense for me to just check for the range strictly between `sessionStartDate` and `sessionEndDate` and have the method named `hasOverlappingSession()`. Let me know what you think.

Reply:
maybe can explain that in the javadoc comment?

Reply:
I have decided to leave it for now, because if I were to remove equality checks, we would have to modify some logic in our test cases. So any test cases that are of the exact same time, should not be used for the test in overlap. A little counterintuitive, but yeah let me know what you think.